test/test_xrandom.cpp

```diff
@@ -167,6 +170,43 @@ namespace xt
         XT_ASSERT_THROW(xt::random::choice(multidim_input, 5, true), std::runtime_error);
     }
 
+    TEST(xrandom, weighted_choice)
+    {
+        xarray<int> a = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+        xarray<double> w = {1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0};
+        xt::random::seed(42);
+        auto ac1 = xt::random::choice(a, 6, w, false);
+        auto ac2 = xt::random::choice(a, 6, w, false);
+        xt::random::seed(42);
+        auto ac3 = xt::random::choice(a, 6, w, false);
+
+        static_assert(std::is_same<decltype(a)::value_type, decltype(ac1)::value_type>::value, "Elements must be same type");
+        ASSERT_EQ(ac1, ac3);
+        ASSERT_NE(ac1, ac2);
+        ASSERT_TRUE(all(isin(ac1, a)));
+        ASSERT_TRUE(all(equal(ac1 % 2, 1)));
+
+        xt::random::seed(42);
+        auto acr1 = xt::random::choice(a, 6, w, false);
+        auto acr2 = xt::random::choice(a, 6, w, false);
+        xt::random::seed(42);
+        auto acr3 = xt::random::choice(a, 6, w, false);
+        ASSERT_EQ(acr1, acr3);
+        ASSERT_NE(acr1, acr2);
+        ASSERT_TRUE(all(isin(acr1, a)));
+        ASSERT_TRUE(all(equal(acr1 % 2, 1)));
+
+        xarray<double> b = {-1, 1};
+        xarray<double> v = {1, 1};
+        xt::random::seed(42);
+        XT_ASSERT_THROW(xt::random::choice(b, 5, v, false), std::runtime_error);
+        XT_ASSERT_NO_THROW(xt::random::choice(b, 5, v, true));
+        xarray<double> multidim_input = { {1,2,3}, {3,4,5} };
+        XT_ASSERT_THROW(xt::random::choice(multidim_input, 5, v, true), std::runtime_error);
+        xarray<double> bad_count_weights = {1, 1, 4};
+        XT_ASSERT_THROW(xt::random::choice(b, 5, bad_count_weights, true), std::runtime_error);
```

Comment:
Not sure that I find this very necessary.

Reply:
Depends on what we do for the exceptions, in the previous `xt::random::choice` with exceptions, they were being tested.